homeassistant/components/influxdb.py

```diff
@@ -89,56 +95,57 @@ def setup(hass, config):
     def influx_event_listener(event):
         """Listen for new messages on the bus and sends them to Influx."""
         state = event.data.get('new_state')
-        if state is None or state.entity_id in blacklist:
-            return
-
-        if whitelist and state.entity_id not in whitelist:
+        if state is None or state.state in (
+                STATE_UNKNOWN, '', STATE_UNAVAILABLE) or \
+                state.entity_id in blacklist:
             return
 
         try:
+            if len(whitelist) > 0 and state.entity_id not in whitelist:
+                return
+
             _state = state_helper.state_as_number(state)
+            _state_key = "value"
         except ValueError:
             _state = state.state
+            _state_key = "state"
+
+        if override_measurement:
+            measurement = override_measurement
+        else:
+            measurement = state.attributes.get('unit_of_measurement')
+            if measurement in (None, ''):
+                if default_measurement:
+                    measurement = default_measurement
+                else:
+                    measurement = state.entity_id
 
-        # Create a counter for this state change
         json_body = [
             {
-                'measurement': "hass.state.count",
+                'measurement': measurement,
                 'tags': {
                     'domain': state.domain,
                     'entity_id': state.object_id,
                 },
                 'time': event.time_fired,
                 'fields': {
-                    'value': 1
+                    _state_key: _state,
```

Comment:
This needs to be cast to a float to prevent inconsistent values that cause data type errors